workflow_manager.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/microsoft/BladeMonRT/nodes"
+	"github.com/microsoft/BladeMonRT/workflows"
+)
+
+/** Class for parsing workflow definitions. */
+type WorkflowManager struct {
+	nameToWorkflow map[string]WorkflowDescription
+}
+
+/** Class for the workflow description in the JSON. */
+type WorkflowDescription struct {
+	Type string `json:"type"`
+	Nodes []NodeDescription `json:"nodes"`
+}
+
+/** Class for the node description in the JSON. */
+type NodeDescription struct {
+	Name string `json:"name"`
+	Metadata interface{} `json:"metadata"`
+	Args interface{} `json:"args"`
+}
+
+func newWorkflowManager(workflowsJson []byte) WorkflowManager {
+	var workflows map[string]map[string]WorkflowDescription
+	json.Unmarshal([]byte(workflowsJson), &workflows)
+
+	return WorkflowManager{nameToWorkflow : workflows["workflows"]}
+}
+
+func (workflowManager *WorkflowManager) constructWorkflow(workflowName string) workflows.InterfaceWorkflow {
+	var currWorkflowDescription WorkflowDescription
+	currWorkflowDescription = workflowManager.nameToWorkflow[workflowName]
+	if currWorkflowDescription.Type != "simple" {
```

Comment:
let's make this a case-switch statement like we have to the NodeRegistry

Reply:
I see, done